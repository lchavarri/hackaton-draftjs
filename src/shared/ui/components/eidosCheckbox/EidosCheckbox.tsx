import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCheck } from '@fortawesome/pro-light-svg-icons';

type Props = {
  checked: boolean;
  onChange: any;
  id?: string;
  label?: string;
  radio?: boolean;
  value?: string | number | undefined;
};

type State = {};

class EidosCheckbox extends React.Component<any, State> {
  renderIcon = (checked: boolean) => {
    if (checked) {
      return <FontAwesomeIcon icon={faCheck} />;
    }
  };

  render() {
    const { checked, onChange, id, label, radio, value } = this.props;
    return (
      <div className={'ui checkbox svg' + (checked ? ' checked' : '')}>
        <input
          type={radio ? 'radio' : 'checkbox'}
          className="hidden"
          tabIndex={0}
          id={id}
          checked={checked}
          onChange={onChange}
          value={value}
        />
        <label htmlFor={id}>
          {this.renderIcon(checked)}
          {label}
        </label>
      </div>
    ); //<Checkbox {...this.props} />
  }
}

export default EidosCheckbox;
