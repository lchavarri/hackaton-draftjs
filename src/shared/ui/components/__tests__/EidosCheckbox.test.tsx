import React from 'react';
import { shallow, mount, ReactWrapper } from 'enzyme';
import EidosCheckbox from '../eidosCheckbox/EidosCheckbox';

describe('EidosCheckbox component', () => {
  let component: ReactWrapper;
  let props: any;

  const defaultProps = {};
  const getComponent = () => <EidosCheckbox {...props} />;
  const wrap = () => shallow(getComponent());
  const wrapDeep = () => mount(getComponent());

  beforeEach(() => {
    props = defaultProps;
    component = wrap();
  });

  it('should match component snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
