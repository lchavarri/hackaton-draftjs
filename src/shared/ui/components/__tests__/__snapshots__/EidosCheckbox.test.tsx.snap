// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EidosCheckbox component should match component snapshot 1`] = `
<div
  className="ui checkbox svg"
>
  <input
    className="hidden"
    tabIndex={0}
    type="checkbox"
  />
  <label />
</div>
`;
